site/app/controllers/DockerInterfaceController.php

```diff
@@ -80,7 +80,7 @@ public function addImage() {
         }
 
         // check for proper format
-        $match = preg_match('/^[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+\/[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+:[a-zA-Z0-9][a-zA-Z0-9._-]{0,127}$/', $_POST['image']);
+        $match = preg_match('/^([a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+\/)?[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+:[a-zA-Z0-9][a-zA-Z0-9._-]{0,127}$/', $_POST['image']);
```

Comment:
Do you know what would happen if an instructor used this container for grading as well? Would it "work" or just error in some way?

The worry I have is that these official images will "work", but because they lack seccomp and stuff, then we're not properly safeguarding execution, and instructors would not realize this.

Reply:
Since the Rust example gradeable runs, I don't think there will be an issue running the containers.  As for seccomp I _think_ that the safeguards will apply since `cargo` fails to run in the Rust container without the `statx` syscall allowed, but I'm not sure what exactly Submitty does to safeguard execution so there might be more constraints which don't get applied to the containers.

What would be the difference between an official image vs a "Submitty-blessed" one?